extensions/http-curl/tests/AlertTests.cpp

```diff
@@ -0,0 +1,148 @@
+    }
+    alerts_.enqueue(std::move(batch));
+    return true;
+  }
+
+  std::string agent_id_;
+  utils::ConditionConcurrentQueue<std::vector<std::string>> alerts_;
+};
+
+class VerifyAlerts : public HTTPIntegrationBase {
+ public:
+  void testSetup() override {}
+
+  void runAssertions() override {
+    verify_();
+  }
+
+  std::function<bool()> verify_;
+};
+
+TEST_CASE("Alert system forwards logs") {
+  auto clock = std::make_shared<utils::ManualClock>();
+  utils::timeutils::setClock(clock);
+
+  TempDirectory dir;
+  auto flow_file = std::filesystem::path(dir.getPath()) / "config.yml";
+  std::ofstream(flow_file) << empty_flow;
+
+  std::string agent_id = "test-agent-1";
+  VerifyAlerts harness;
+  AlertHandler handler(agent_id);
+  harness.setUrl("http://localhost:0/api/alerts", &handler);
+  harness.getConfiguration()->set(minifi::Configuration::nifi_c2_agent_identifier, agent_id);
+  harness.getConfiguration()->setHome(dir.getPath());
+
+  auto log_props = std::make_shared<logging::LoggerProperties>();
+  log_props->set("appender.alert1", "alert");
+  log_props->set("appender.alert1.url", harness.getC2RestUrl());
+  log_props->set("appender.alert1.filter", ".*<begin>(.*)<end>.*");
+  log_props->set("appender.alert1.rate.limit", "10 s");
+  log_props->set("appender.alert1.flush.period", "1 s");
+  log_props->set("logger.root", "INFO,alert1");
+  logging::LoggerConfiguration::getConfiguration().initialize(log_props);
+
+  auto verifyLogsArrived = [&] (const std::vector<std::string>& expected) {
+    std::vector<std::string> logs;
+    REQUIRE(handler.alerts_.dequeueWaitFor(logs, 1s));
+    REQUIRE(logs.size() == expected.size());
+    for (size_t idx = 0; idx < expected.size(); ++idx) {
+      bool contains = std::search(logs[idx].begin(), logs[idx].end(), expected[idx].begin(), expected[idx].end()) != logs[idx].end();
+      REQUIRE(contains);
+    }
+  };
+
+  harness.verify_ = [&] {
+    auto logger = logging::LoggerFactory<minifi::FlowController>::getLogger();
+    // time = 0
+    logger->log_error("not matched");
+    logger->log_error("<begin>one<end>");
+    logger->log_error("not the same but treated so <begin>one<end>");
+    logger->log_error("<begin>two<end>");
+    clock->advance(2s);
+    // time = 2
+    verifyLogsArrived({
+        "<begin>one<end>", "<begin>two<end>"
+    });
+
+    clock->advance(5s);
+    // time = 7
+    // no new logs over HTTP
+
+    logger->log_error("other <begin>one<end>");
+    logger->log_error("new log <begin>three<end>");
+    clock->advance(2s);
+
+    // time = 9
+    verifyLogsArrived({
+        "new log <begin>three<end>"
+    });
+
+    clock->advance(2s);
+    // time = 11
+    logger->log_error("other <begin>one<end>");
+    logger->log_error("new log <begin>three<end>");
+    clock->advance(2s);
+    // time = 13
+
+    verifyLogsArrived({
+        "other <begin>one<end>"
+    });
+
+    return true;
+  };
+
+  harness.run(flow_file, dir.getPath());
```

Comment:
```suggestion
  harness.run(flow_file.string(), dir.getPath());
```
as `std::filesystem::path` doesn't convert implicitly to `std::string` on Windows.

Reply:
done